### PR TITLE
CI build and release pipelines

### LINKED ISSUE #75: GitHub CI build and release pipeline
Write a CI pipeline GitHub workflow to build and release the application APK on GitHub on a button press.

### PR DESCRIPTION
**Pull request by gruvw**
This PR aims to improve the continuous integration/delivery capabilities on this project.

Fixes #75 

### Completed tasks

- [X] GitHub release action
    - [x] Build APK from CI
    - [X] Upload file with renamed APK
    - [X] Create release from action with file
    - [X] Correct version from YAML (with release title)
    - [X] Git tag
    - [X] Standard commit changelog integration
    - [X] Release on push of a button (`workflow_dispatch`)
- [X] Cleaned other GitHub actions
- [X] Test coverage badge on README
- [X] Workflow to check that APK can be built correctly + upload APK to artifact

### Features

We can now see the coverage on the main branch directly on top of the README (wait 2 minutes for update).

We can also press a button in the **Actions** tap on the GitHub repository and then press the **Flutter Application GitHub Release** and **Run workflow**.
Be careful with this as it creates a git tagged GitHub Release and automatically generates the CHANGELOG using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
It is intended to be triggered only once per sprint (for the demo) and we need to bump the application version number in the `pubspec.yaml` file **before** making a new release (on any PR at the start of the sprint).

We now also check that the application APK is correctly generated on every push (so only push when necessary) and it is uploaded to the build workflow artifacts, so anyone can later download the generated APK for any particular point in the history.
This last "build" workflow takes a little bit of time to complete (about 5 min) but it is not necessary to wait for completion as for >99% of cases it should complete successfully if the "testing" workflow finishes successfully.

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -15,23 +15,40 @@ jobs:
...
+        run: flutter analyze
 
-      - name: Flutter testing
+      - name: Flutter Run Tests (coverage)
         run: |
           dart pub global activate dlcov
           dlcov gen-refs
           flutter test --coverage -r github test
-          dlcov -c 80 --exclude-files "*.g.dart,*firebase_options.dart"
+          set -o pipefail
```
Comment (TheTexanCodeur) [inline .github/workflows/tests.yaml:35]: good idea to make the pipeline fail if any command fails!

File: scripts/coverage [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #!/usr/bin/env bash
 
+set -e
```
Comment (TheTexanCodeur) [inline scripts/coverage:3]: Again, good idea to make the script exit directly after an error! This helps ensuring that the script doesn't continue executing in an unexpected state.

**Review by TheTexanCodeur — APPROVED** [review]
Good job! 

These new github actions automate a lot of work and are thus really helpful. The other changes in the existing actions also make the code easier to read and more consistent. 

The commands that you added to stop scripts that fail also add a layer of security by ensuring that the action fails directly if anything goes wrong.

Again, i think that you made great job! LGTM :)

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      - name: Flutter Build APK
+        run: ./scripts/build
+
+      - name: Get Version From Project
+        run: echo "VERSION=$(grep 'version:' pubspec.yaml | awk '{print $2}')" >> $GITHUB_ENV
+
+      - name: Setup git-cliff Changelog
+        uses: kenji-miyake/setup-git-cliff@v1
+
+      - name: Generate Changelog
```
Comment (camillelnne) [inline .github/workflows/release.yaml:35]: Great idea, using of git cliff for the changelog !

**Review by camillelnne — APPROVED** [review]
Great features ! Very thorough.
LGTM :)

**Pull request merged**